packages/ui/src/theme/createTheme/resolveObject.ts

```diff
@@ -0,0 +1,169 @@
+// Internal Style Dictionary methods
+// copied from amzn/style-dictionary with the owner's permission
+import {
+  cloneDeep,
+  createReferenceRegex,
+  getName,
+  getPathFromName,
+  GroupMessages,
+  has,
+  resolveReference,
+  usesReference,
+} from '../../utils';
+
+const PROPERTY_REFERENCE_WARNINGS =
+  GroupMessages.GROUP.PropertyReferenceWarnings;
+
+let current_context = []; // To maintain the context to be able to test for circular definitions
```

Comment:
Singletons scare me, does this need to be one? Same comment for all module scoped data structures

Reply:
Updated to remove all module-level variables.